Simplified/NYPLReadiumBookmark.swift

```diff
@@ -101,14 +101,19 @@
   override func isEqual(_ object: Any?) -> Bool {
     let other = object as! NYPLReadiumBookmark
 
-    if (self.contentCFI == other.contentCFI &&
-      self.idref == other.idref &&
-      self.chapter == other.chapter &&
-      self.location == other.location)
-    {
-      return true
+    if let contentCFI = self.contentCFI,
+      let otherContentCFI = other.contentCFI,
+      contentCFI.count > 0 && otherContentCFI.count > 0 {
+        // R1
+        return self.idref == other.idref
+            && self.contentCFI == other.contentCFI
+            && self.location == other.location
```

Comment:
for R1 this test should also include `self.chapter == other.chaper`? (according to previous code)

actually maybe for R2 too? (not sure)